### PR TITLE
Implement `use` argument for LLM for decorators

### PR DESCRIPTION
**Pull request by cw75**
## Describe your changes and why you are making these changes
This PR implements the `use` argument that allows the user to specify the intent to use our LLM library. Under the hood, we include the `use_llm` flag into the resource struct and use this as a flag to select Docker images that have LLM Python library dependencies and Cuda drivers pre-installed.

## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: sdk/aqueduct/__init__.py [hunk 1]
```diff
@@ -7,6 +7,8 @@
 from aqueduct.flow import Flow
 from aqueduct.schedule import DayOfMonth, DayOfWeek, Hour, Minute, daily, hourly, monthly, weekly
 
+llm = "llm"
```
Comment (hsubbaraj-spiral) [inline sdk/aqueduct/__init__.py:10]: Can we declare this as a constant?

File: src/golang/lib/models/shared/operator/operator.go [hunk 2]
```diff
@@ -47,6 +47,7 @@ type ResourceConfig struct {
 	MemoryMB        *int               `json:"memory_mb,omitempty"`
 	GPUResourceName *string            `json:"gpu_resource_name,omitempty"`
 	CudaVersion     *CudaVersionNumber `json:"cuda_version,omitempty"`
+	UseLlm          *bool              `json:"use_llm,omitempty"`
```
Comment (hsubbaraj-spiral) [inline src/golang/lib/models/shared/operator/operator.go:50]: Change to `UseLLM` just to be consistent with our naming scheme.

**Review by hsubbaraj-spiral — APPROVED** [review]
Few small nits, but otherwise looks great!

File: sdk/aqueduct/__init__.py (hunk 1 above)
Comment (cw75) [reply sdk/aqueduct/__init__.py]: removed this as it's not needed

**Comment by cw75** [comment]
@hsubbaraj-spiral updated the implementation to add the `llm_op` API shown in [this](https://www.loom.com/share/28f7388e677e42f4a7b3ed1419b1da74) demo. If you can take a look at `llm_wrapper.py` that'd be great.

File: sdk/aqueduct/llm_wrapper.py [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+
+        module = __import__("aqueduct_llm", fromlist=[llm_name])
+        llm = getattr(module, llm_name)
+
+        if "prompt" in parameters:
+            prompt = parameters["prompt"]
+            if not isinstance(prompt, str):
+                raise ValueError("The 'prompt' parameter must be a string.")
+
+            if "{text}" not in prompt:
```
Comment (hsubbaraj-spiral) [inline sdk/aqueduct/llm_wrapper.py:59]: Can you explain the significance of `"{text}"` is this a keyword in LLM prompts?

**Review by hsubbaraj-spiral — APPROVED** [review]
Yep, the new API makes sense to me, just a quick questions about the prompt.

File: sdk/aqueduct/llm_wrapper.py (hunk 3 above)
Comment (cw75) [reply sdk/aqueduct/llm_wrapper.py:59]: "If the prompt contains {text}, we will replace {text} with the input string(s) before sending to the LLM. If the prompt does not contain {text}, we will prepend the prompt to the input string(s) before sending to the LLM."

**Pull request merged**